public/app/features/alerting/unified/components/rule-editor/ConditionField.tsx

```diff
@@ -29,9 +29,20 @@ export const ConditionField: FC = () => {
     [queries]
   );
 
+  const expressions = useMemo(() => {
+    return queries.filter((query) => query.datasourceUid === ExpressionDatasourceUID);
+  }, [queries]);
+
+  // automatically use the last expression when new expressions have been added
+  useEffect(() => {
```

Comment:
Doesn't this interfere with the `useEffect` below? It's a bit difficult to understand why we need them both. Could they be unified somehow?